vars/publishReports.groovy

```diff
@@ -8,57 +8,67 @@
 def call(List<String> files, Map params = [:]) {
     def timeout = params.get('timeout') ?: '60'
 
-    if (!infra.isTrusted()) {
+    if (!infra.isTrusted() && !infra.isInfra()) {
         error 'Can only call publishReports from within the trusted.ci environment'
     }
 
     withCredentials([
         string(credentialsId: 'azure-reports-access-key', variable: 'AZURE_STORAGE_KEY'),
     ]) {
-        docker.image('mcr.microsoft.com/azure-cli:2.0.41').inside {
-            for(int i = 0; i < files.size(); ++i) {
-                String filename = files[i]
-                withEnv(['HOME=/tmp']) {
-                    String uploadFlags = ''
-                    switch (filename) {
-                        case ~/(?i).*\.html/:
-                            uploadFlags = '--content-type="text/html"'
-                            break
-                        case ~/(?i).*\.css/:
-                            uploadFlags = '--content-type="text/css"'
-                            break
-                        case ~/(?i).*\.json/:
-                            uploadFlags = '--content-type="application/json"'
-                            break
-                        case ~/(?i).*\.js/:
-                            uploadFlags = '--content-type="application/javascript"'
-                            break
-                        case ~/(?i).*\.gif/:
-                            uploadFlags = '--content-type="image/gif"'
-                            break
-                        case ~/(?i).*\.png/:
-                            uploadFlags = '--content-type="image/png"'
-                            break
-                    }
-                    // Blob container can be removed once files are uploaded on the azure file storage
-                    sh "az storage blob upload --account-name=prodjenkinsreports --container=reports --timeout=${timeout} --file=${filename} --name=${filename} ${uploadFlags}"
+        // Install the az CLI if not already present. TODO: remove once CLI installed in universal template image
+        sh '''
+        if [ ! command -v az >/dev/null 2>&1];
+        then
+            apk add --no-cache gcc musl-dev python3-dev libffi-dev openssl-dev cargo make \
+            && apk add --no-cache py3-pip py3-pynacl py3-cryptography \
+            && python3 -m pip install --no-cache-dir azure-cli
```

Comment:
this is probably downloading around 400mb or so each build so I wouldn't keep this for long if you can avoid it

Reply:
yep, I've added until the `jenkins-infra/docker-helmfile` with the az CLI is available for the build. Thanks for the tip

Reply:
removed